apps/mobile-wallet/src/features/addressesManagement/AddressDetailsModal.tsx

```diff
@@ -0,0 +1,136 @@
+import AnimatedBackground from '~/components/AnimatedBackground'
+import BalanceSummary from '~/components/BalanceSummary'
+import ActionCardButton from '~/components/buttons/ActionCardButton'
+import RoundedCard from '~/components/RoundedCard'
+import BottomModal from '~/features/modals/BottomModal'
+import { closeModal, openModal } from '~/features/modals/modalActions'
+import withModal from '~/features/modals/withModal'
+import usePersistAddressSettings from '~/hooks/layout/usePersistAddressSettings'
+import { useAppDispatch, useAppSelector } from '~/hooks/redux'
+import RootStackParamList from '~/navigation/rootStackRoutes'
+import { SendNavigationParamList } from '~/navigation/SendNavigation'
+import { addressSettingsSaved, makeSelectAddressesTokens, selectAddressByHash } from '~/store/addressesSlice'
+import { VERTICAL_GAP } from '~/style/globalStyle'
+import { Address } from '~/types/addresses'
+import { showToast, ToastDuration } from '~/utils/layout'
+
+export interface AddressDetailsModalProps {
+  addressHash: AddressHash
+}
+
+const AddressDetailsModal = withModal<AddressDetailsModalProps>(({ id, addressHash }) => {
+  const { t } = useTranslation()
+  const navigation = useNavigation<NavigationProp<SendNavigationParamList | RootStackParamList>>()
+  const dispatch = useAppDispatch()
+
+  const address = useAppSelector((s) => selectAddressByHash(s, addressHash))
+  const selectAddressTokens = useMemo(makeSelectAddressesTokens, [])
+  const addressTokens = useAppSelector((s) => selectAddressTokens(s, addressHash))
+
+  const handleSendPress = () => {
+    sendAnalytics({ event: 'Address modal: Pressed btn to send funds from' })
+
+    navigation.navigate('SendNavigation', {
+      screen: 'DestinationScreen',
+      params: { fromAddressHash: addressHash }
+    })
+
+    dispatch(closeModal({ id }))
+  }
+
+  const handleReceivePress = () => {
+    sendAnalytics({ event: 'Address modal: Pressed btn to receive funds to' })
+
+    dispatch(openModal({ name: 'ReceiveQRCodeModal', props: { addressHash } }))
+  }
+
+  const handleSettingsPress = () => {
+    dispatch(openModal({ name: 'AddressSettingsModal', props: { addressHash, parentModalId: id } }))
+  }
+
+  return (
+    <BottomModal modalId={id} title={<AddressBadge addressHash={addressHash} fontSize={16} />}>
+      <Content>
+        <RoundedCard>
+          <AnimatedBackground shade={address?.settings.color} isAnimated />
+          <BalanceSummary addressHash={addressHash} />
+        </RoundedCard>
+
+        <ActionButtons>
+          {addressTokens.length > 0 && (
+            <ActionCardButton title={t('Send')} onPress={handleSendPress} iconProps={{ name: 'send' }} />
+          )}
+          <ActionCardButton title={t('Receive')} onPress={handleReceivePress} iconProps={{ name: 'download' }} />
+          <ActionCardButton title={t('Settings')} onPress={handleSettingsPress} iconProps={{ name: 'settings' }} />
+          <SetDefaultAddressCardButton address={address} />
```

Comment:
nit: By passing the whole address object then once the default address changes the whole `AddressDetailsModal` will rerender. That's ok, but we did not benefit from my suggestion earlier. The best way would be to not get the `address` object in the modal and only pass `addressHash` to its subcomponents. I implemented this in this PR so that you don't have to, feel free to include it by merging: https://github.com/alephium/alephium-frontend/pull/1112